fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1196,7 +1237,12 @@ private ScoredPlan planOrderedUnion(@Nonnull PlanContext planContext, @Nonnull L
         if (unionPlan.getComplexity() > complexityThreshold) {
             throw new RecordQueryPlanComplexityException(unionPlan);
         }
-        return new ScoredPlan(1, unionPlan, Collections.emptyList(), anyDuplicates, includedRankComparisons);
+
+        // If we don't change this when shouldAttemptFailedInJoinAsOr() is true, then we _always_ pick the union plan,
+        // rather than the in join plan.
+        int score = getConfiguration().shouldAttemptFailedInJoinAsOr() ? 0 : 1;
```

Comment:
Hmm, interesting. Should the logic in the join filter that chooses between the `Union` and the `InJoin` be changed then (maybe ignoring the score if we can't trust it?).

Reply:
Possibly. I'm a bit worried about doing so because I don't really know what the `score` is trying to capture. If we think that's lower risk, I'm for it.